### ISSUE TITLE
Fix issue in pickRandom returning given array instead of random picks when the given number is greater than its length

### ISSUE THREAD

**Issue by josdejong** [josdejong/mathjs#1990]
The current implementation of `pickRandom` will normally return a set of random values from the provided options, but there is one exception: when you provide say 10 values and you request 10 or more random picks from it, the function will not actually do anything but return the original array.

Originally this was intended code, there is a unit test for it, but I can't imagine why `pickRandom` would return _non_ random results in some cases.

It is this part in the code:

```js
    if (length === 0) {
      return []
    } else if (number >= length) {
      return number > 1 ? possibles : possibles[0]
    }
```

Since this can be considered a change in behavior, we'll schedule this change for `v8`.

Ref: #1976

**Comment by fweth** [comment]
Actually I had situation where I wanted the current behaviour.  For example, when I have an array of points, each point specifed as a 2-array containing x-position and y-position, I actually expect pickRandom to return a point, not the position of a random point along a random axis.  Maybe we could have multidimensional arrays being tagged with an extra property (like `arr.dimensions=2`) to distinguish them from e.g. a list of vectors?

**Comment by josdejong** [comment]
Sorry to hear that @fweth. 

I suppose you refer to the change in `pickRandom` published in [`mathjs@7.5.0`](https://github.com/josdejong/mathjs/blob/develop/HISTORY.md#2020-10-07-version-750), not the change proposed in this issue of fixing returning a random value when requesting more random numbers than provided? I think this "feature" was an unintended side effect, but I'm not sure (there where no unit tests in place at least to cover this behavior).

Good idea to think this through a bit more. Introducing an option is a good idea. Maybe `elementWise: boolean` or `flatten: boolean` or something like that.

**Comment by fweth** [comment]
Oh sorry, you are right, I commented on the wrong issue.  My bad, I'll delete the comment :)

**Comment by josdejong** [comment]
ow, but can you please open your comment in a new issue then? It's good to get it addressed I think?

**Comment by josdejong** [comment]
@fweth in e788c1d7f336f5a3dd9662a01ec65453eaad219a I've introduced a new option `elementWise` for the function `pickRandom`, so you can do something like:

```
pickRandom([{x: 1.0, y: 2.0}, {x: 1.1, y: 2.0}], { elementWise: false })
```

which will then return an array item.

Does this address your use case?

**Comment by fweth** [comment]
Yes, that addresses my use case!  Thank you!

**Comment by josdejong** [comment]
Fix is published now in `v8.0.0`
